syntax = "proto3";

service DualToRActive {
  rpc QuerySide(SideRequest) returns (SideReply) {}
  rpc QueryAdminForwardingPortState(AdminRequest) returns (AdminReply) {}
  rpc SetAdminForwardingPortState(AdminRequest) returns (AdminReply) {}
  rpc QueryOperationPortState(OperationRequest) returns (OperationReply) {}
  rpc QueryLinkState(LinkStateRequest) returns (LinkStateReply) {}
}

message AdminRequest {
  repeated int32 portid = 1;
  repeated bool state = 2;
}

message AdminReply {
  repeated int32 portid = 1;
  repeated bool state = 2;
}

message OperationRequest {
  repeated int32 portid = 1;
}

message OperationReply {
  repeated int32 portid = 1;
  repeated bool state = 2;
}

message SideRequest {
  int32 portid = 1;   // valid port id, 0 or 1. In the future, 
                      // we can extend to have more values.
}

message SideReply {
  int32 side = 1;     // valid side id, 0 or 1. In the future, 
                      // we can extend to have more values.
}

message LinkStateRequest {
  repeated int32 portid = 1;
}

message LinkStateReply {
  repeated int32 portid = 1;
  repeated bool state = 2;
}




